core/src/main/scala/kafka/controller/ControllerChannelManager.scala

```diff
@@ -443,7 +443,8 @@ abstract class AbstractControllerBrokerRequestBatch(config: KafkaConfig,
 
   private def sendLeaderAndIsrRequest(controllerEpoch: Int, stateChangeLog: StateChangeLogger): Unit = {
     val leaderAndIsrRequestVersion: Short =
-      if (config.interBrokerProtocolVersion >= KAFKA_2_2_IV0) 2
+      if (config.interBrokerProtocolVersion >= KAFKA_2_3_IV1) 3
```

Comment:
The new LeaderAndIsr version was introduced with KIP-455, which would come out in 2.4. I believe we need to check against that version
https://github.com/apache/kafka/pull/7351#issuecomment-532267852